cmd/update/volume_move.go

```diff
@@ -0,0 +1,383 @@
+
+	if !isMovingMaster {
+		// mark the replica on the [source] node as preferred for eviction
+		//
+		// the master node needs to first be moved and then be evicted due to
+		// api endpoints restriction
+		params := &apiclient.SetPreferredEvictionCandidatesRequestParams{
+			DeploymentIDs: []string{deploymentOnSourceNode.String()},
+		}
+		if c.useCAS() {
+			params.CASVersion = vol.Version
+		}
+		err = c.client.SetPreferredEvictionCandidates(ctx, nsID.String(), vol.ID.String(), params)
+		if err != nil {
+			return fmt.Errorf("failed to set preferred eviction candidates: %s", err)
+		}
+
+		c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica %s on [source] node %s marked as preferred for eviction", deploymentOnSourceNode, c.fromNode))
+	}
+
+	// wait for the new replica on [destination] node to sync
+	err = c.waitForReplicaOnNodeToSync(ctx, nsID)
+	if err != nil {
+		return err
+	}
+
+	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [destination] node %s finished syncing", c.toNode))
+
+	if !isMovingMaster {
+		_ = c.client.EvictReplica(ctx, nsID.String(), vol.ID.String(), deploymentOnSourceNode.String())
+		if err != nil {
+			if !errors.Is(err, apiclient.VolumeNotFoundError{}) {
+				return fmt.Errorf("failed to force evict replica from [source] node %s (will eventually converge): %w", c.fromNode, err)
+			}
+		}
+
+		c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [source] node %s successfully evicted", c.fromNode))
+		return nil
+	}
+
+	// from here on, we are attempting to move a master
+
+	// find the deployment IDs of the newly created replica and the master being moved
+	var newReplicaID id.Deployment
+	for _, rep := range vol.Replicas {
+		if rep.Node.String() == c.toNode {
+			newReplicaID = rep.ID
+		}
+	}
+	oldMasterID := vol.Master.ID
+
+	err = c.client.AttemptPromotion(ctx, string(nsID), vol.ID.String(), newReplicaID.String())
+	if err != nil {
+		return err
+	}
+
+	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [destination] node %s attempting promotion to master", c.toNode))
+
+	err = c.waitForMasterPromote(ctx, nsID)
+	if err != nil {
+		return err
+	}
+
+	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [destination] node %s successfully promoted to master", c.toNode))
+
+	err = c.client.EvictReplica(ctx, nsID.String(), vol.ID.String(), oldMasterID.String())
+	if err != nil {
+		if !errors.Is(err, apiclient.VolumeNotFoundError{}) {
+			return fmt.Errorf("failed to force evict old master from [source] node %s (will eventually converge): %w", oldMasterID.String(), err)
+		}
+	}
+
+	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("old master on [source] node %s successfully evicted", c.toNode))
+	return nil
+}
+
+func (c *volumeMoveCommand) waitForReplicaOnNode(ctx context.Context, nsID id.Namespace) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timed out while watching for desired state but the operation may still be running (server side)")
```

Comment:
It's worth adding the reason the ctx is done. It's almost always going to be `context.DeadlineExceeded`, but it may be `context.Cancelled`, which is useful debugging info.  
```
fmt.Errorf("timed out while watching for desired state but the operation may still be running (server side) - %w", ctx.Err())
```

Reply:
Updated the message as suggested